### PR TITLE
Skip automatic removal of missing media from workpiece, when no media…

### PR DESCRIPTION
**Pull request by markusweigelt**
… was found, Fix some potential npes

Fixes issue #5579

### CODE REVIEW

**Comment by markusweigelt** [comment]
@henning-gerhardt 

> At least, there should be a message that includes an explicit request for the user to delete the references. If this is not confirmed, the references remain. This message must be worded very clearly to avoid accidental wrong selection.

With the current implementation of this pr, the references are not remove and only a warning message is shown what happend.
Is it necessary to show a dialog to select loading behaviour.  I think it can be very annoying and may also lead to user errors. What do you think? Does the problem occur not so often and the selection errors are negligible (cause you can reload metadataeditor if you select the wrong loading behaviour if the user notices it)?

**Comment by andre-hohmann** [comment]
@markusweigelt : Thanks a lot!

A warning without a dialog function is absolutely sufficient. If someone needs a dialog function, this can be developed supplementary. 

Unfortunately, it is difficult to estimate how often certain cases occur in the respective facilities. The discussion following  https://github.com/kitodo/kitodo-production/issues/5579#issuecomment-1460308477 confirms in my opinion, that your suggestion is the best way to solve the basic problem (removal of needed references) with little effort. 

Especially https://github.com/kitodo/kitodo-production/issues/5579#issuecomment-1466264052 shows that there are probably no use cases in which the removal of references is needed.

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 1]
```diff
@@ -292,12 +292,18 @@ public void setSelectedPhysicalNode(TreeNode selectedPhysicalNode) {
     }
 
     Optional<LogicalDivision> getSelectedStructure() {
+        if (Objects.isNull(selectedLogicalNode)) {
+            return null;
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java:296]: According to Coding Guidelines returning `null` value should be avoided. There are a few places in the source code where an additionl null check must be added and not only at the one place later in this pull request.

According to my IDE (IntelliJ IDEA) the return statement here should be written as `return Optional.empty()`.

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 2]
```diff
@@ -292,12 +292,18 @@ public void setSelectedPhysicalNode(TreeNode selectedPhysicalNode) {
...
+        }
         StructureTreeNode structureTreeNode = (StructureTreeNode) selectedLogicalNode.getData();
         Object dataObject = structureTreeNode.getDataObject();
         return Optional.ofNullable(dataObject instanceof LogicalDivision ? (LogicalDivision) dataObject : null);
     }
 
     Optional<PhysicalDivision> getSelectedPhysicalDivision() {
+        if (Objects.isNull(selectedPhysicalNode)) {
+            return null;
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java:305]: See comment above.

File: Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java [hunk 3]
```diff
@@ -203,13 +204,13 @@ private boolean executeOtherScript(List<Process> processes, String script)
                 copyDataToChildren(processes, script);
                 break;
             default:
-                return executeRemainingScript(processes, script);
+                return executeRemainingScript(processes);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java:207]: Change is not related to the issue. Please revert this and open a separate pull request for this change.

File: Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java [hunk 4]
```diff
@@ -203,13 +204,13 @@ private boolean executeOtherScript(List<Process> processes, String script)
...
         }
         return true;
     }
 
-    private boolean executeRemainingScript(List<Process> processes, String script)
-            throws DataException, IOException, InvalidImagesException {
+    private boolean executeRemainingScript(List<Process> processes)
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java:212]: See comment above. Removing of the unused parameter is a separate issue.

File: Kitodo/src/main/java/org/kitodo/production/services/dataeditor/DataEditorService.java [hunk 5]
```diff
@@ -242,7 +242,7 @@ public static List<SelectItem> getAddableMetadataForMediaUnit(DataEditorForm dat
      */
     public static StructuralElementViewInterface getStructuralElementView(DataEditorForm dataEditor) {
         Optional<LogicalDivision> selectedStructure = dataEditor.getSelectedStructure();
-        if (selectedStructure.isPresent()) {
+        if (Objects.nonNull(selectedStructure) && selectedStructure.isPresent()) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/dataeditor/DataEditorService.java:245]: The null check could be avoided if the `getSelectedStructure()` method is changed in the already mentioned way (`return null` -> `return Optional.empty()`.

**Review by henning-gerhardt — CHANGES_REQUESTED** [review]
This is a code only review. Functionality check will be performed later. Maybe adding tests should help here and prevent changes in future in a bad way.

**Comment by markusweigelt** [comment]
> Especially [#5579 (comment)](https://github.com/kitodo/kitodo-production/issues/5579#issuecomment-1466264052) shows that there are probably no use cases in which the removal of references is needed.

Maybe it makes more sense to take out the deletion directly than to introduce this distinction. Alternatively we could also add a project option to clean up the file references which is turned off by default or a button in metadataeditor to do it manually.

Who decides here if this is still needed?

**Comment by andre-hohmann** [comment]
@markusweigelt : Thanks a lot for the PR. I had a look on it and it works!
The only thing I noticed is that when the language is set to English, the message in German language is displayed. 
It's not "bad" and maybe it's just because of our configuration. Nevertheless, I want to draw attention to it.

**Review by henning-gerhardt — APPROVED** [review]
Changes looks nice and clean and adding a test is very nice.

Why the German warn message appears but the interface is set in English is unclear and maybe not related to the changes itself.

**Pull request merged**